plc4j/protocols/driver-bases/base/src/main/java/org/apache/plc4x/java/base/connection/AbstractPlcConnection.java

```diff
@@ -43,15 +38,18 @@ Licensed to the Apache Software Foundation (ASF) under one
  */
 public abstract class AbstractPlcConnection implements PlcConnection, PlcConnectionMetadata {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractPlcConnection.class);
-
-    private static final int PING_TIMEOUT_MS = 1_000;
-
     @Override
     public PlcConnectionMetadata getMetadata() {
         return this;
     }
 
+    @Override
+    public CompletableFuture<Void> ping() {
+        CompletableFuture<Void> future = new CompletableFuture<>();
+        future.completeExceptionally(new PlcUnsupportedOperationException("The connection does not support pinging"));
```

Comment:
I would suggest to use https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/CompletableFuture.html#failedFuture(java.lang.Throwable) here if possible.

Reply:
Argh ... was looking for something like that ... strangely I must have missed it :-) ... of course ... will update that.

Reply:
Ha! ... I knew there was something ... if I used the failedFuture method, we would have to bump up the Java level to 9 which I wasn't planning on doing just for this one convenience method.

Reply:
fair enough. :)